### PR TITLE
esp32s3/wifi: Add Wi-Fi driver (STA mode) for ESP32-S3

### PR DESCRIPTION
**Pull request by tmedicci**
## Summary

Implement Wi-Fi (STA mode) support on ESP32-S3 (NUT-663)

1) Wi-Fi driver libs from Espressif ESP-IDF release/v5.0;
2) Station mode only;
3) WPA2-PSK and WPA3-SAE enabled;

Not yet supported (WIP):
- SoftAP mode;
- 802.11k, 802.11v and 802.11R;
- Power Save mode;
- BLE Coexistence;

## Impact

Add Wi-Fi driver on ESP32-S3.

## Testing

Internal CI (`wapi` defconfig) + `iperf` long-term testing.

### CODE REVIEW

**Review by acassis — APPROVED** [review]

File: arch/xtensa/src/esp32s3/esp32s3_systemreset.h [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+#ifndef __ARCH_XTENSA_SRC_ESP32S3_ESP32S3_SYSTEMRESET_H
+#define __ARCH_XTENSA_SRC_ESP32S3_ESP32S3_SYSTEMRESET_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdint.h>
+
+#ifndef __ASSEMBLY__
```
Comment (pkarashchenko) [inline arch/xtensa/src/esp32s3/esp32s3_systemreset.h:30]: If I recall correctly the stdint.h should also be hidden under this ifdef (but maybe things changed from last time I tried it).

File: arch/xtensa/src/esp32s3/esp32s3_wifi_utils.h [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ * Included Files
+ ****************************************************************************/
+
+#ifndef __ARCH_XTENSA_SRC_ESP32S3_ESP32S3_WIFI_UTILS_H
+#define __ARCH_XTENSA_SRC_ESP32S3_ESP32S3_WIFI_UTILS_H
+
+#include <nuttx/config.h>
+#include <nuttx/net/netdev.h>
+
+#include <stdint.h>
```
Comment (pkarashchenko) [inline arch/xtensa/src/esp32s3/esp32s3_wifi_utils.h:31]: ditto

File: arch/xtensa/src/esp32s3/esp32s3_systemreset.h (hunk 1 above)
Comment (tmedicci) [reply arch/xtensa/src/esp32s3/esp32s3_systemreset.h:28]: You're right! Done!

File: arch/xtensa/src/esp32s3/esp32s3_wifi_utils.h (hunk 2 above)
Comment (tmedicci) [reply arch/xtensa/src/esp32s3/esp32s3_wifi_utils.h]: You're right! Done!

**Pull request merged**